src/karma.rs

```diff
@@ -1,177 +1,291 @@
+    fn default() -> KarmaStyle {
+        KarmaStyle::Implicit
+    }
 }
 
 impl KarmaStyle {
     pub fn most_explicit(first: KarmaStyle, second: KarmaStyle) -> KarmaStyle {
-        if first == KarmaStyle::Implicit && second == KarmaStyle::Implicit {
-            KarmaStyle::Implicit
-        } else if first == KarmaStyle::Silent && second != KarmaStyle::Notify {
-            KarmaStyle::Silent
-        } else if first == KarmaStyle::Notify {
-            KarmaStyle::Notify
-        } else {
-            second
+        match (first, second) {
+            (KarmaStyle::Implicit, KarmaStyle::Implicit) => KarmaStyle::Implicit,
+            (KarmaStyle::Silent, KarmaStyle::Notify) => KarmaStyle::Silent,
+            (KarmaStyle::Notify, _) => KarmaStyle::Notify,
+            (_, second) => second,
         }
     }
 }
 
 #[derive(Debug, Clone, PartialEq)]
-pub struct Karma {
+pub struct KarmaChange {
     pub term: String,
-    pub change: KarmaChange,
-    pub style: KarmaStyle
+    pub votes: KarmaAmount,
+    pub style: KarmaStyle,
 }
 
-#[derive(Debug, Clone, PartialEq)]
-pub struct KarmaValue {
+impl KarmaChange {
+    pub fn new(term: &str, votes: KarmaAmount, style: KarmaStyle) -> KarmaChange {
+        KarmaChange {
+            term: canonicalize_term(term),
+            votes,
+            style,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum Aliases {
+    To(String),
+    FromOther(Vec<String>),
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Karma {
     pub term: String,
     pub original_term: String,
-    pub votes: KarmaChange,
-    pub last_vote: DateTime<Local>,
-    pub first_vote: DateTime<Local>
+    pub votes: KarmaAmount,
+    pub last_vote: Option<DateTime<Local>>,
+    pub first_vote: Option<DateTime<Local>>,
+    pub aliases: Option<Aliases>,
+}
+
+impl std::fmt::Display for Karma {
+    fn fmt(&self, w: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(w, "{}", self.term)
+    }
 }
 
-impl KarmaValue {
-    pub fn new(term: &str) -> KarmaValue {
-        KarmaValue {
-            term: term.to_ascii_lowercase(),
+impl Karma {
+    pub fn new(term: &str) -> Karma {
+        let now = Local::now();
+        Karma {
+            term: canonicalize_term(term),
             original_term: term.to_string(),
-            votes: KarmaChange::new(0, 0),
-            last_vote: Local::now(),
-            first_vote: Local::now()
+            votes: KarmaAmount::default(),
+            last_vote: Some(now),
+            first_vote: Some(now),
+            aliases: None,
         }
     }
 
-    pub fn vote(&mut self, karma: &Karma) {
-        self.last_vote = Local::now();
-        self.votes.up += karma.change.up;
-        self.votes.down += karma.change.down;
+    pub fn vote(&mut self, karma: &KarmaAmount) {
+        self.last_vote = Some(Local::now());
+        self.votes.up += karma.up;
+        self.votes.down += karma.down;
```

Comment:
Maybe implement `std::ops::Add` for `KarmaAmount` instead?

Reply:
Ack. Will do.